### DISCUSSION TITLE
`VPN_PORT_FORWARDING_LISTENING_PORT` vs `FIREWALL_VPN_INPUT_PORTS`

### DISCUSSION THREAD

**Discussion by SinTan1729** [qdm12/gluetun#2028] [Q&A]
I've using `FIREWALL_VPN_INPUT_PORTS` to get port forwarding in a custom setup. But the latest release mentions `VPN_PORT_FORWARDING_LISTENING_PORT`. I couldn't find out what it is in the Wiki. It's mentioned in a few issues, and it sound like it does the same thing as `FIREWALL_VPN_INPUT_PORTS` i.e. give me a static forwarded port.

So, what are the differences between these? Or is `FIREWALL_VPN_INPUT_PORTS` being deprecated?

**Comment by sockenklaus** [comment]
I am just guessing but just now I was configuring port forwarding and had trouble getting a connection with `FIREWALL_VPN_INPUT_PORTS`.
I changed the Environment variable to `VPN_PORT_FORWARDING_LISTENING_PORT` and port forwarding started working. So maybe you're right and the former variable has been deprecated.

Would appreciate a confirmation of a dev though.

**Reply by SinTan1729** [reply]
We're probably doing different things. For me, `FIREWALL_VPN_INPUT_PORTS` is the one that works.

**Reply by sockenklaus** [reply]
Oh i just checked my logs again:

When using `FIREWALL_VPN_INPUT_PORTS` gluetun logs:

````
2024-03-19T22:39:36+01:00 INFO [firewall] setting allowed input port xxxx through interface tun0...
````
This log entry doesn't appear when NOT using this variable. I don't know why it appeared  that forwarding worked only when using `VPN_PORT_FORWARDING_LISTENING_PORT`...
